include/arch/dma_structs.hpp

```diff
@@ -299,7 +299,7 @@ struct dma_small_object {
 private:
 	dma_pool *_pool;
 	T *_data;
-	std::aligned_storage_t<sizeof(T), alignof(T)> _embedded;
+	alignas(alignof(T)) std::byte _embedded[sizeof(T)];
```

Comment:
Please also change `&_embedded` -> `_embedded` in the constructor (for readability, doesn't make a difference in semantics).